### PR TITLE
Support dstack

### PR DESCRIPTION
**Pull request by fukatani**
This PR is for supporting `functions.dstack` mentioned in #1280 .

### CODE REVIEW

File: chainer/functions/array/dstack.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        if xs[0].ndim <= 2:
+            ys = xp.dsplit(gy[0], len(xs))
+            return [y.reshape(xs[0].shape) for y in ys]
+        else:
+            sizes = numpy.array([x.shape[2] for x in xs[:-1]]).cumsum()
+            return xp.dsplit(gy[0], sizes)
+
+
+def dstack(xs):
+    """Concatenate variables along third axis (deep wise).
```
Comment (muupan) [inline chainer/functions/array/dstack.py]: Ditto.

File: docs/source/reference/functions.rst [hunk 2]
```diff
@@ -111,6 +111,10 @@ copy
 ~~~~
 .. autofunction:: copy
 
+dstack
+~~~~
```
Comment (muupan) [inline docs/source/reference/functions.rst]: Six tildes are needed here.

**Comment by muupan** [comment]
LGTM except the comment above.

**Comment by fukatani** [comment]
Thanks for your review! :smiley: 
I fixed them.

**Comment by muupan** [comment]
Thanks!

**Pull request merged**